### ISSUE TITLE
Possible NULL pointer dereference in hunspell.cxx:616

### ISSUE THREAD

**Issue by phajdan** [hunspell/hunspell#241]
Visual Studio code analysis picked this up, [a C6011](https://msdn.microsoft.com/en-us/library/2ayc37ac.aspx).

In function pipe_interface, at the line:

```
i->word = mystrdup(buf+1);
```

....Hunspell dereferences `i`, which was `malloc`'d the line before, as:

```
struct wordlist* i = 
(struct wordlist *) malloc (sizeof(struct wordlist));
```

....and never checked against NULL. Sure, it's a rare condition, but a valid one nonetheless.

```
C6011   Dereferencing NULL pointer 'i'.
    614 'i' may be NULL
    616 'i' is dereferenced, but may still be NULL
```

Original comment by: ariccio

Original Ticket: [hunspell/bugs/274](https://sourceforge.net/p/hunspell/bugs/274)

**Comment by ariccio** [comment]
Self-xref: @ariccio

**Comment by dimztimz** [comment]
Fixed in 43802938b3bc00d398d35c7782a64a6517a50e77
